### PR TITLE
Mixed patches

### PR DESCRIPTION
**Pull request by GavinHuttley**
## Summary by Sourcery

Refactor moltype handling by introducing a dedicated literal type alias, consolidating parameter typing, and standardizing validation and error messaging across the codebase while updating related tests to reflect the new behavior.

Enhancements:
- Introduce MolTypeLiteral alias and update moltype parameter typing across core and app modules
- Unify get_moltype signature to accept literals or MolType instances, handle None input with a warning fallback, and simplify lookup
- Replace internal moltype validation asserts with explicit MolTypeError/ValueError and improved error messages
- Add and refine type hints for functions such as intersection, union, get_distance_calculator, and from_array_names

Tests:
- Update tests to pass explicit moltype to make_seq, expect ValueError instead of AssertionError, and add tests for get_moltype(None) warning and EbiParser warning capture

### CODE REVIEW

File: src/cogent3/app/translate.py [hunk 1]
```diff
@@ -350,14 +350,14 @@ def __init__(
         s1    MR
         s2    .-
         """
-        moltype = cogent3.get_moltype(moltype)
-        assert moltype.label.lower() in ("dna", "rna"), "Invalid moltype"
```
[the remark this answers is not in the document]
Comment (GavinHuttley) [reply src/cogent3/app/translate.py:354]: You're hallucinating. That code is being deleted.
